### PR TITLE
Add gprc-plugin operator docs

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

### CODE REVIEW

**Review by jpkrohling — COMMENTED** [review]
LGTM, but would be good to have an OK from @JStickler as well.

File: content/docs/next-release/operator.md [hunk 1]
```diff
@@ -674,6 +674,47 @@ spec:
 
 <2> Disables creation of default index templates.
 
+### Storage plugin
+
+Storage type `grpc-plugin` allows using Jaeger with 3rd party storage implementations.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply content/docs/next-release/operator.md:679]: Setting `spec.storage.type` to `grpc-plugin` enables using Jaeger with 3rd party storage implementations.

File: content/docs/next-release/operator.md [hunk 2]
```diff
@@ -674,6 +674,47 @@ spec:
...
+
+Follows an example of Jaeger CR with `allInOne` strategy and Clickhouse storage plugin.
```
Comment (JStickler) [inline content/docs/next-release/operator.md:681]: The following is an example of a Jaeger CR using the `allInOne` deployment strategy and the Clickhouse storage plugin.  Refer to [jaeger-clickhouse](https://github.com/pavolloffay/jaeger-clickhouse) for documentation and a fully functional example.

File: content/docs/next-release/operator.md [hunk 3]
```diff
@@ -674,6 +674,47 @@ spec:
...
+  volumeMounts:
+    - name: plugin-config
+      mountPath: /plugin-config
+  volumes:
+    - name: plugin-config
+      configMap:
+        name: jaeger-clickhouse # <4>
+ ```
+
+<1> `grpc-plugin` storage type.
```
Comment (JStickler) [inline content/docs/next-release/operator.md:710]: Storage type set to `grpc-plugin`

File: content/docs/next-release/operator.md [hunk 4]
```diff
@@ -674,6 +674,47 @@ spec:
...
+
+<2> Image with the plugin binary. The image is used as init-container to copy the binary into volume that is available to the Jaeger process. The image has to copy the binary into `/plugin` directory.
+
+<3> Configuration options for the `grpc-plugin`.
+
+<4> Config map with the plugin configuration.
```
Comment (JStickler) [inline content/docs/next-release/operator.md:716]: Question, does the user need to create this config map?  Or is it created automatically?  I'm not sure why you're calling it out here.

**Review by JStickler — COMMENTED** [review]
Couple of suggestions, one question.

File: content/docs/next-release/operator.md (hunk 4 above)
Comment (pavolloffay) [reply content/docs/next-release/operator.md:716]: bc this is the only supported way to create a config for the plugin. I have rephrased the sentence to be clear that user is responsible for creating it.

**Comment by pavolloffay** [comment]
PR updated. @jpkrohling or @JStickler could you please approve?

File: content/docs/next-release/operator.md (hunk 2 above)
Comment (jpkrohling) [reply content/docs/next-release/operator.md:681]: @pavolloffay are you using @JStickler's version for this paragraph?

Comment (pavolloffay) [reply content/docs/next-release/operator.md]: now yes, I didn't push this one before

**Review by JStickler — APPROVED** [review]
LGTM

File: content/docs/next-release/operator.md (hunk 4 above)
Comment (JStickler) [reply content/docs/next-release/operator.md]: That helps clarify, thanks!

**Comment by pavolloffay** [comment]
thanks @JStickler

**Pull request merged**